radixdlt-java/src/main/java/com/radixdlt/client/application/translate/DataStoreTranslator.java

```diff
@@ -40,7 +40,7 @@ public Completable translate(DataStore dataStore, AtomBuilder atomBuilder) {
 		Map<String, Object> metaData = new HashMap<>();
 		metaData.put("timestamp", atom.getTimestamp());
 		metaData.put("signatures", atom.getSignatures());
-		metaData.put("application", atom.getDataParticle().getApplication());
+		metaData.put("application", atom.getDataParticle().getMetaData("application"));
```

Comment:
I'm not sure this is ideal - it will create a `[key, null]` pair in the map, which is probably not desirable for something that is missing, as calls to `containsKey(...)` will return true, but `get(...)` will return `null`.

Could use something like `metaData.compute("application", (k, v) -> atom.getDataParticle().getMetaData("application"))` which will not add a key with a `null` value.

Reply:
good point, that works.

Reply:
though this logic should be refactored down the line, so that inserting a null value isn't exposed.